v0/destinations/ga4/utils.js

```diff
@@ -284,6 +284,29 @@ function getDestinationItemProperties(message, isItemsRequired) {
   return items;
 }
 
+// This function will return exclusion list for a particular event.
+// Exclusion list= sourceKeys that are already mapped
+function getExclusionList(mappingJson) {
+  const ga4ExclusionList = [];
+
+  mappingJson.forEach(element => {
+    const mappingSourceKeys = element.sourceKeys;
+
+    if (typeof mappingSourceKeys === "string") {
+      ga4ExclusionList.push(mappingSourceKeys.split(".").pop());
+    } else {
+      mappingSourceKeys.forEach(item => {
+        ga4ExclusionList.push(item.split(".").pop());
+      });
+    }
+  });
+
+  // We are mapping "products" to "items", so to remove redundancy we should not send products again
+  ga4ExclusionList.push("products");
```

Comment:
We also need to add `GA4_RESERVED_PARAMETER_EXCLUSION` list here as reserved parameter names cannot be used. This list is present above.

Reply:
Done